src/main/java/org/commcare/session/CommCareSession.java

```diff
@@ -195,19 +176,19 @@ public String getNeededData(EvaluationContext evalContext) {
             return SessionFrame.STATE_COMMAND_ID;
         }
 
-        Vector<Entry> remainingValidEntries =
-                getEntriesForCommand(currentCmd, collectedDatums, true);
-        String needDatum = getDataNeededByAllEntries(remainingValidEntries);
+        Vector<Entry> entriesForCurrentCommand = getEntriesForCommand(currentCmd);
+        String needDatum = getDataNeededByAllEntries(entriesForCurrentCommand);
 
         if (needDatum != null) {
             return needDatum;
-        } else if (remainingValidEntries.isEmpty()) {
-            throw new RuntimeException("Collected datums don't match required datums for entries at command " + currentCmd);
-        } else if (remainingValidEntries.size() == 1
-                && remainingValidEntries.elementAt(0) instanceof RemoteRequestEntry
-                && ((RemoteRequestEntry)remainingValidEntries.elementAt(0)).getPostRequest().isRelevant(evalContext)) {
+        } else if (entriesForCurrentCommand.isEmpty()) {
+            // No entries available directly within the current command, so we must need to select another menu
+            return SessionFrame.STATE_COMMAND_ID;
```

Comment:
This is the new change: If the current command has no entries, instead of throwing an exception, we assume that this is just because there no entries at this level but there are some nested in child modules